### ISSUE TITLE
oauth2: improve design of RsaJwsSignature 

### ISSUE THREAD

**Issue by mgenov** [clouway/oauth2-server#48]
The existing RsaJwsSignature was designed just for verification but as extension of the library to support signing (id tokens) it requires re-design. 

```java
public class RsaJwsSignature implements com.clouway.oauth2.jws.Signature {
  private byte[] signature;

  public RsaJwsSignature(byte[] signature) {
    this.signature = signature;
  }

  public boolean verifyWithPrivateKey(byte[] content, Pem.Block privateKey) {
     ...
  }
  @Override
  public boolean verify(byte[] content, Block key) {
      ..
  }

  @Override
  public String sign(byte[] content, Pem.Block privateKey) {
      ... 
  }
}
```

The Constructor accepts signature  but in signing process it's not required as it's the result. This is why the design of sign/verify need to be improved.

**Comment by mgenov** [comment]
The design is going to be changed with adding of #50

**Comment by VasilMitovClouway** [comment]
Fixed in #51
